src/control/server/server.go

```diff
@@ -87,6 +90,46 @@ func iommuDetected() bool {
 	return len(dmars) > 0
 }
 
+// detectVmd returns whether VMD devices have been found and a slice of VMD
+// PCI addresses if found.
+func detectVmd() ([]string, error) {
+	// Check available VMD devices with command:
+	// "$lspci | grep  -i -E "201d | Volume Management Device"
+	lspciCmd := exec.Command("lspci")
+	vmdCmd := exec.Command("grep", "-i", "-E", "201d|Volume Management Device")
```

Comment:
@sydneyvanda instead of running lspci, couldn't we instead run spdk discover with spdk_init_vmd() and only set enable vmd if we find controllers with pci type "vmd"? I doubt  calling init_vmd() will cause problems if run on system with no VMD as its just a library call. we could then derive VMD parent addresses from the domain of the backing device PCI address.

Reply:
I agree with the parsing text, but if just seemed the simplest method at the time. @tanabarr  I think just running SPDK discover should work, or basically doing something similar to app/spdk_lspci/spdk_lspci.c. I'm not sure of anything other than lspci to find the vmd address, scanning /sys/bus/pci/devices will show the pci address of the VMD capable NVMe (1000*) but I'm not sure that is very useful in this case